### ISSUE TITLE
"Collection was modified" in CollectAndSerializeAsync()

### ISSUE THREAD

**Issue by gboucher90** [prometheus-net/prometheus-net#464]
Hi,

We are using prometheus-net 8.2.0 and we are observing occasionally on all our services (and all the envs: staging, preprod, prod) those kind of errors:
```
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at System.Collections.Generic.Dictionary`2.ValueCollection.Enumerator.MoveNext()
   at Prometheus.CollectorRegistry.UpdateRegistryMetrics()
   at Prometheus.CollectorRegistry.CollectAndSerializeAsync(IMetricsSerializer serializer, CancellationToken cancel)
   at Prometheus.MetricServerMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.InvokeCore(HttpContext context, PathString matchedPath, PathString remainingPath)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
```

It seems to happen very rarely per instance (between 1 and 3 times), an instance can live between a few hours and a few days.

I'm unclear if this is a recent regression or we never noticed it before. I would think it could come from recent 8.2.0 performance improvements as it touches this path:
https://github.com/prometheus-net/prometheus-net/commit/9f0752301601d0b33de33639509f3b8716f74cf2#diff-92eed7c273df734f3fe9010cb841f28a3a855a8933857f646478c2d9c5949fe2

I'm not sure there is anything custom with our code, we have a pretty standard and simple usage of the lib. We rely on default `Metrics.DefaultRegistry` and have maybe ~ 30 metrics per service, most of them being counter or histograms with some labels.

Thank you!

**Comment by sandersaares** [comment]
Thanks for reporting. I will take a look.

**Comment by sandersaares** [comment]
Indeed, this seems to be a regression in 8.2.0. Fixed in 8.2.1, should be available on nuget.org shortly.

**Comment by gboucher90** [comment]
> Indeed, this seems to be a regression in 8.2.0. Fixed in 8.2.1, should be available on nuget.org shortly.

Thanks a lot 🙏 
Will try to deploy this within the next few days and will keep you updated if it fixed all the occurrences.

**Comment by gboucher90** [comment]
We didn't had a single occurrence (in any of our envs) since the upgrade to 8.2.1 and it has been more than 24h so I think this is indeed fixed.
Thank you again for the very quick fix!
